libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -49,6 +50,15 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
       }
 
       vote = std::make_shared<Vote>(vote_with_block_rlp[0]);
+      vote_hash = vote->getHash();
+
+      // Synchronization point in case multiple threads are processing the same vote at the same time
+      if (!seen_votes_.insert(vote_hash)) {
+        LOG(log_tr_) << "Received vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
+                     << ") already seen.";
+        continue;
+      }
```

Comment:
it is not worth it to put it here. The code is not clean right now but if we enter this if branch:
```
    // There is included also PBFT block as optional value in Vote packet
    if (packet_data.rlp_.itemCount() == 1 && packet_data.rlp_[0].itemCount() == kVotePacketWithBlockSize &&
        packet_data.rlp_[0][0].isList()) {
```

there is only single vote so parsing 1 vote does not delay marking it as seen